shared/video-sdk/develop/audio-and-voice-effects/project-implementation/windows.mdx

```diff
@@ -0,0 +1,232 @@
+
+### Handle the system logic
+
+This section describes the steps required to set up access to the UI elements.
+
+1. **Define variables to manage audio effects and access the UI elements**
+
+    In `AgoraImplementationDlg.h`, add the following declarations to the `CAgoraImplementation` class:
+
+    ```cpp
+	int soundEffectId = 1; // Unique identifier for the sound effect file
+	std::string soundEffectFilePath = "https://www.soundjay.com/human/applause-01.mp3"; // URL or path to the sound effect
+	int soundEffectStatus = 0;
+	int voiceEffectIndex = 0;
+	boolean audioPlaying = false; // Manage the audio mixing state
+	std::string audioFilePath = "https://www.kozco.com/tech/organfinale.mp3"; // URL or path to the audio mixing file
+	CButton* playEffectButton, *voiceEffectButton, *audioMixingButton;
+	CButton* speakerphoneSwitch;
+    ```
+
+1. **Set up access to the speakerphone switch**
+
+    In `AgoraImplementationDlg.cpp`, add the following lines to the `OnInitDialog` method after `setupVideoSDKEngine();`
+
+    <ProductWrapper product = "interactive-live-streaming">
+    ```cpp
+    audioMixingButton = (CButton*)GetDlgItem(IDC_BUTTON3);
+    audioMixingButton->SetWindowText(L"Mix Audio");
+	playEffectButton = (CButton*)GetDlgItem(IDC_BUTTON4);
+    playEffectButton->SetWindowText(L"Audio Effect");
+	voiceEffectButton = (CButton*)GetDlgItem(IDC_BUTTON5);
+    voiceEffectButton->SetWindowText(L"Voice Effect");
+    ```
+    </ProductWrapper>
+    <ProductWrapper product = "video-calling">
+    ```cpp
+    speakerphoneSwitch = (CButton*)GetDlgItem(IDC_CHECK1);
+    audioMixingButton = (CButton*)GetDlgItem(IDC_BUTTON3);
+    audioMixingButton->SetWindowText(L"Mix Audio");
+	playEffectButton = (CButton*)GetDlgItem(IDC_BUTTON4);    
+    playEffectButton->SetWindowText(L"Audio Effect");
+	voiceEffectButton = (CButton*)GetDlgItem(IDC_BUTTON5);
+    voiceEffectButton->SetWindowText(L"Voice Effect");
+    ```
+    </ProductWrapper>
+
+### Implement sound and voice effects
+
+To add sound and voice effect to your <Vpl k="CLIENT" />, take the following steps:
+
+1. **Enable the user to start and stop audio mixing**
+        
+    When the user presses **Mix Audio**, the <Vg k = "VSDK"/> fetches an audio file using `audioFilePath` and mixes it with the local audio stream. The remote and local users listen the audio file playing with the audio stream. To enable audio mixing in your <Vpl k = "CLIENT"/>, in **Dialog Editor**, double-click **Mix Audio**. **Dialog Editor** automatically creates and opens an event listener for you. Add the following code to the event listener you just created:
+      
+    ```cpp
+    audioPlaying = ~audioPlaying;
+	if (audioPlaying) 
+    {
+		audioMixingButton->SetWindowText(L"Stop Mixing");
+		agoraEngine->startAudioMixing(audioFilePath.c_str(), false, 1, 0);
+	}
+	else
+	{
+		agoraEngine->stopAudioMixing();
+		audioMixingButton->SetWindowText(L"Mix Audio");
+	}
+    ```
+
+1. **Pre-load sound effects**
+
```

Comment:
Google style guide advises to avoid the hyphen and use a closed form of the verb, unless it is one of the exceptions: https://developers.google.com/style/hyphens